### ISSUE TITLE
RejectedExecutionException in AppStateCollector

### ISSUE THREAD

**Issue by pkovalenko** [SiftScience/sift-android#33]
SiftScience version to 0.9.11.
On ALCATEL  5054, Samsung S7, Samsung SM-G950U, LG V10 (Android 5, Android 7, Android 4), I see crashes with next stacktrace.

```
Fatal Exception: java.util.concurrent.RejectedExecutionException: Task java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask@3b9bc0a4 rejected from java.util.concurrent.ScheduledThreadPoolExecutor@38db2e0d[Shutting down, pool size = 1, active threads = 0, queued tasks = 1, completed tasks = 2]
       at java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2011)
       at java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:793)
       at java.util.concurrent.ScheduledThreadPoolExecutor.delayedExecute(ScheduledThreadPoolExecutor.java:298)
       at java.util.concurrent.ScheduledThreadPoolExecutor.schedule(ScheduledThreadPoolExecutor.java:503)
       at java.util.concurrent.Executors$DelegatedScheduledExecutorService.schedule(Executors.java:644)
       at siftscience.android.AppStateCollector.onConnected(AppStateCollector.java:250)
       at com.google.android.gms.common.internal.zzad.zzake(Unknown Source)
       at com.google.android.gms.common.api.internal.zzbd.zzd(Unknown Source)
       at com.google.android.gms.common.api.internal.zzar.zzahc(Unknown Source)
       at com.google.android.gms.common.api.internal.zzar.begin(Unknown Source)
       at com.google.android.gms.common.api.internal.zzbl.onConnected(Unknown Source)
       at com.google.android.gms.common.api.internal.zzw.onConnected(Unknown Source)
       at com.google.android.gms.common.internal.zzab.onConnected(Unknown Source)
       at com.google.android.gms.common.internal.zzn.zzj(Unknown Source)
       at com.google.android.gms.common.internal.zze.zzajn(Unknown Source)
       at com.google.android.gms.common.internal.zzi.zzajo(Unknown Source)
       at com.google.android.gms.common.internal.zzh.handleMessage(Unknown Source)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:135)
       at android.app.ActivityThread.main(ActivityThread.java:5422)
       at java.lang.reflect.Method.invoke(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:372)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:914)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:707)
```

Could you please take a look at this issue. Thank you!

**Comment by quanpod** [comment]
Acknowledging the issue, @garylee1 to take the initial look and then we'll get something on our backlog. Thanks @pkovalenko

**Comment by garylee1** [comment]
Thanks @pkovalenko we'll get the fix out for this soon. It should only be happening on shutdown, so there's not a user-facing impact.

**Comment by garylee1** [comment]
@pkovalenko This has been fixed and released with v0.9.12. Thanks for your patience!
